### ISSUE TITLE
Network returning invalid probability values

### ISSUE THREAD

**Issue by floydsoft** [fchollet/keras#319]
Hi,

I'm doing a binary classification task, the `predict_proba` prompts `Network returning invalid probability values.` warning sometimes, are there any tips to mitigate this problem?

Thanks!

**Comment by vzhong** [comment]
Can you post some code to reproduce this? Also, would you happen to be able to print out the inputs to `predict_proba` and the outputs of `predict_proba`?

**Comment by floydsoft** [comment]
Hi, the model configured as below:

```python
    rnn = Sequential()
    rnn.add(Dropout(0.5))
    rnn.add(JZS3(63, 128))
    rnn.add(BatchNormalization((128,)))
    rnn.add(MaxoutDense(128, 64, nb_feature=512))
    rnn.add(BatchNormalization((64,)))
    rnn.add(Dropout(0.5))

    mlp = Sequential()
    mlp.add(Dropout(0.5))
    mlp.add(Dense(26, 64, init='glorot_uniform', W_regularizer=l2(0.01)))
    mlp.add(PReLU((64,)))
    mlp.add(BatchNormalization((64,)))
    mlp.add(MaxoutDense(64, 32, nb_feature=256))
    mlp.add(BatchNormalization((32,)))
    mlp.add(Dropout(0.5))

    model = Sequential()
    model.add(Merge([rnn, mlp], mode='concat'))

    model.add(Dense(96, 1, init='glorot_uniform', W_constraint=nonneg(), W_regularizer=l2(0.01)))
    model.add(PReLU((1,)))

    model.compile(
        loss='mean_squared_error',
        optimizer='adam',
        class_mode="binary"
    )
```

The input tensor are huge, might not paste to here trivially...

**Comment by floydsoft** [comment]
All values in the tensor are non-negative, some in [0, 1].

**Comment by vzhong** [comment]
it seems like your output is not normalized to a probability distribution. Namely, your final layer is formed by rectifiers, which are not bounded. That is to say, they are not valid probability values. I would imagine that is why you are seeing the warning.

**Comment by floydsoft** [comment]
@vzhong nice answer! I'll try it, thank you!

**Comment by vzhong** [comment]
@floydsoft you're welcome :)

In addition, should you decide to normalize the output to a probability distribution, I would suggest changing your loss function to 'binary_crossentropy' as opposed to 'mean_squared_error'.

**Comment by floydsoft** [comment]
@vzhong ok, I'll try it :+1:

**Comment by fchollet** [comment]
If you are actually looking for probabilities, you should replace the last PReLU with a sigmoid Activation. If you are not looking for probabilities, you should use the method predict() instead of predict_proba().

**Comment by floydsoft** [comment]
@fchollet yep, it's normal now.
